### PR TITLE
Adds {proxy+} support in local lambdas

### PR DESCRIPTION
**Pull request by palasjir**
- Adds support for local dev server to support AWS Lambda Proxy Integrations in API Gateway.
  That Allows to set following configuration:
  ```
  lambdaConfigs: [
    {
      publicPath: '/bff/{proxy+}',
      httpMethod: 'ANY'
      ...
    }
  ]
  ```

- Adds some missing `APIGatewayProxyEvent` properties

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -12,7 +12,8 @@ export type LambdaHttpMethod =
   | 'OPTIONS'
   | 'PATCH'
   | 'POST'
-  | 'PUT';
+  | 'PUT'
+  | 'ANY';
```
Comment (clebert) [inline src/types.ts]: I have only recently removed the Catch-ANY functionality. In my memory an ANY route on AWS had not worked. Were you able to test this successfully with a real deployed stack?

See:
https://github.com/clebert/aws-simple/commit/f24ae9a6f722040cff3180bfd6df773425f5b51e

File: src/express/utils/create-lambda-request-handler.ts [hunk 2]
```diff
@@ -30,8 +30,9 @@ export function createLambdaRequestHandler(
           event: {
             ...getRequestHeaders(req),
             path: req.path,
+            httpMethod: req.method,
             queryStringParameters: req.query,
-            body: req.body
+            body: req.body ? JSON.stringify(req.body) : null
```
Comment (clebert) [inline src/express/utils/create-lambda-request-handler.ts]: Is this a bug in the current implementation? After a short research the body in the event of a Lambda proxy integration seems to be indeed a string.

@unstubbable Can you confirm that?

Comment (unstubbable) [reply src/express/utils/create-lambda-request-handler.ts]: Since the text parser is used, req.body is already a string. Did this not work in your case?

Comment (palasjir) [reply src/express/utils/create-lambda-request-handler.ts]: I was experiencing `body` being passed to lambda as empty object. But looking at the type of `APIGatewayProxyEvent` it should only be `string | null`.

File: src/types.ts (hunk 1 above)
Comment (palasjir) [reply src/types.ts]: You're right it doesn't not work with deployment. I will remove that part.

File: src/express/utils/create-lambda-request-handler.ts (hunk 2 above)
Comment (clebert) [reply src/express/utils/create-lambda-request-handler.ts]: An additional Stringify may even lead to an invalid result. What would be if I get a JSON string which I then stringify again. I would then have double quotation marks. Wouldn't it be better to check if it is already a string, if not then use Stringify or null.

Comment (palasjir) [reply src/express/utils/create-lambda-request-handler.ts]: Thats a very valid point. I didn't realised that `JSON.stringify` adds additional quotes. I changed it to be used for non-strings only.

**Pull request merged**